app/layersproxymodel.h

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************/
+
+
+#ifndef LAYERSPROXYMODEL_H
+#define LAYERSPROXYMODEL_H
+
+#include <QObject>
+
+#include "qgsmaplayerproxymodel.h"
+#include "qgsmaplayermodel.h"
+#include "qgsvectorlayer.h"
+#include "qgsmaplayer.h"
+#include "qgsproject.h"
+#include "qgslayertree.h"
+#include "qgslayertreelayer.h"
```

Comment:
some #includes could be surely moved to .cpp and/or replaced by forward declarations?